src/node_sqlite.cc

```diff
@@ -114,10 +117,13 @@ inline void THROW_ERR_SQLITE_ERROR(Isolate* isolate, const char* message) {
 DatabaseSync::DatabaseSync(Environment* env,
                            Local<Object> object,
                            DatabaseOpenConfiguration&& open_config,
-                           bool open)
+                           bool open,
+                           bool allow_load_extension)
     : BaseObject(env, object), open_config_(std::move(open_config)) {
   MakeWeak();
   connection_ = nullptr;
+  allow_load_extension_ = allow_load_extension;
+  enable_load_extension_ = allow_load_extension;
```

Comment:
Do we really need both of these flags. They seem to overlap quite a bit.

Reply:
Im doing this because im thinking only allow you set allowExtension on constructor level, and disallow you enable it when you disable extension.

i saw one comment from community because of security reasons https://github.com/nodejs/node/pull/53900#issuecomment-2282272071